docs/GDPs/gdp-duck-storage.rst

```diff
@@ -0,0 +1,456 @@
+alignment in the generated code is then based on the :code:`backend` parameter of the :code:`stencil` decorator.
+This will continue to work, but in case the storage passed at call-time uses other settings than the backend's default
+settings, these must also be specified to the stencil. We propose the following arguments for the :code:`Field`
+annotation, which are specified using the notation (:code:`Argument[value]`):
+
+:code:`DType`
+   correspoinds to the `dtype` argument
+:code:`LayoutMap`
+   corresponds to the `layout_map` argument
+:code:`Alignment`
+   corresponds to the `alignment` argument
+:code:`DefaultParameters`
+   corresponds to the `default_parameters` argument.
+   Either :code:`'F'` for FORTRAN layout, :code:`'C'` for C/C++-layout or one of the backend identifier strings.
+:code:`Axes`
+   corresponds to the `axes` argument can be a string or as it is now one of :code:`I`, :code:`J`, :code:`K`,
+   :code:`IJ` :code:`IK`, :code:`JK`, :code:`IJK`
+
+The dtype is required, all others optional. The dtype and axes can be specified as positional arguments or using the
+bracket notation, while all others have to be specified using the bracket notation. If any parameter is specified both
+explicitly and in the default parameter set, the explicit value takes precedence. All symbols, including the `Axes`
+arguments can be imported from :code:`gt4py.gtscript`.
+
+.. note::
+   While the storage constructors take the `gpu` argument, it is not necessary to declare this in the stencil
+   signature. The compute device is a property of the backend and can not be set on a per-field basis. If a storage
+   with only a CPU (GPU) buffer is passed to a stencil which is computed on GPU (CPU), an exception is raised.
+
+Examples
+========
+
+For a single-precision 3d field which was allocated in FORTRAN without taking further care about alignment, a simple
+copy-stencil could then read:
+
+.. code-block:: python
+
+   import numpy as np
+   from gt4py import gtscript
+   from gtscript import Field, DefaultParameters
+
+   FieldAnnotation = Field[np.float32, DefaultParameters['F']]
+
+   @gtscript.stencil(backend="debug")
+   def copy(field_in: FieldAnnotation, field_out: FieldAnnotation):
+       field_out[...] = field_in
+
+For a storage which is compatible with the default layout of the :code:`"gtmc"` backend, the annotation could instead
+be defined as :code:`FieldAnnotation = Field[DType[np.float32], Alignment:[8], LayoutMap[(0, 2, 1)]]`.
+However, if the backend actually is :code:`backend="gtmc"`, the following will continue to work:
+:code:`FieldAnnotation = Field[np.float32]`
+
+.. note::
+    Both currently and with the implementation of this GDP, fields with masked axes can be specified. However, since
+    they are not supported in the analysis and code generated yet, we decided to not enable this here yet either,
+    but it shall be part of a later GDP.
```

Comment:
With what purpose? Is this blocking their use? They should be used right now, I do not see why we don't.

Reply:
On one side, the GDP is about storages, and I tried to keep the changes to other parts minimal. 

We can alternatively expand this GDP more and instead do both of the things, but put them in separate implementation steps, where the implementation of the lower-dim'l storages in the DSL would just happen later.

Reply:
The DSL part has been removed from the GDP altogether, it is only about storages now.